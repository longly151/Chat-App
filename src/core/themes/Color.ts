const Color = {
  white: '#ffffff',
  grey0: '#e1e8ee',
  grey1: '#bdc6cf',
  grey2: '#86939e',
  grey3: '#5e6977',
  grey4: '#43484d',
  grey5: '#393e42',
  black: '#121212',

  red: '#e60023',
  orange: '#F2711C',
  yellow: '#FBBD08',
  olive: '#B5CC18',
  green: '#51d96c',
  teal: '#00B5AD',
  blue: '#2185D0',
  violet: '#6435C9',
  purple: '#A333C8',
  pink: '#E03997',
  brown: '#A5673F',

  /**
   * Dark Color
   */
  darkPrimary: '#1EC19A',
  darkSecondary: '#43484d',
  darkSuccess: '#51d96c',
  darkWarning: '#FBBD08',
  darkError: '#e60023',
  darkPrimaryBackground: '#121212',
  darkSecondaryBackground: '#43484d',
  darkPrimaryText: '#ffffff',
  darkSecondaryText: '#86939e',
  darkLoading: '#86939e',

  /**
   * Light Color
   */
  lightPrimary: '#107E83',
  lightSecondary: '#393e42',
  lightSuccess: '#51d96c',
  lightWarning: '#FBBD08',
  lightError: '#e60023',
  lightPrimaryBackground: '#FFFFFF',
  lightSecondaryBackground: '#bdc6cf',
  lightPrimaryText: '#121212',
  lightSecondaryText: '#5e6977',
  lightLoading: '#43484d',
};

export default Color;
