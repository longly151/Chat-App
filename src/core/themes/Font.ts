const fontFamily = {
  Roboto: 'Roboto',
  GoogleSans: 'GoogleSans',
};

const fontSize = {
  tiny: 10,
  small: 12,
  medium: 14,
  large: 18,
  xLarge: 20,
};

const Font = {
  fontFamily,
  fontSize
};
export default Font;
