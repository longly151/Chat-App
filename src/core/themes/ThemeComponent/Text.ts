import Color from '../Color';

/**
 * Text
 */
const headerFontSize = 20;
const xTitleFontSize = 18;
const titleFontSize = 16;
const paragraphFontSize = 14;
const subtitleFontSize = 12;

export const darkText = {
  header: {
    color: Color.darkPrimaryText,
    fontSize: headerFontSize,
  },
  xTitle: {
    color: Color.darkPrimaryText,
    fontSize: xTitleFontSize,
  },
  title: {
    color: Color.darkPrimaryText,
    fontSize: titleFontSize,
  },
  paragraph: {
    color: Color.darkPrimaryText,
    fontSize: paragraphFontSize,
  },
  subtitle: {
    color: Color.darkSecondaryText,
    fontSize: subtitleFontSize,
  },
  style: {
    color: Color.darkPrimaryText,
  }
};
export const lightText = {
  header: {
    color: Color.lightPrimaryText,
    fontSize: headerFontSize,
  },
  xTitle: {
    color: Color.lightPrimaryText,
    fontSize: xTitleFontSize,
  },
  title: {
    color: Color.lightPrimaryText,
    fontSize: titleFontSize,
  },
  paragraph: {
    color: Color.lightPrimaryText,
    fontSize: paragraphFontSize,
  },
  subtitle: {
    color: Color.lightSecondaryText,
    fontSize: subtitleFontSize,
  },
  style: {
    color: Color.lightPrimaryText,
  }
};
