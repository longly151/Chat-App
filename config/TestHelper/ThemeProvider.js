import React from 'react';
import PropTypes from 'prop-types';
import deepmerge from 'deepmerge';
import {lightTheme} from '../../../../src/core/themes/Theme';

export const ThemeContext = React.createContext({
  theme: lightTheme
});

export default class ThemeProvider extends React.Component {
  constructor(props) {
    super(props);

    this.defaultTheme = lightTheme;
    this.state = {
      theme: this.defaultTheme,
      useDark: props.useDark,
    };
  }

  static getDerivedStateFromProps(props, state) {
    const {useDark} = props;
    if (useDark !== state.useDark) {
      return {
        theme: lightTheme,
        useDark,
      };
    }
    return null;
  }

  updateTheme = (updates) => {
    this.setState(({theme}) => ({
      theme: deepmerge(theme, updates),
    }));
  };

  replaceTheme = (theme) => {
    this.setState(() => ({
      theme: deepmerge(this.defaultTheme, theme),
    }));
  };

  getTheme = () => this.state.theme;

  render() {
    return (
      <ThemeContext.Provider
        value={{
          theme: this.state.theme,
          updateTheme: this.updateTheme,
          replaceTheme: this.replaceTheme,
        }}
      >
        {this.props.children}
      </ThemeContext.Provider>
    );
  }
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node.isRequired,
  useDark: PropTypes.bool,
};

ThemeProvider.defaultProps = {
  theme: {},
  useDark: false,
};

export const ThemeConsumer = ThemeContext.Consumer;
