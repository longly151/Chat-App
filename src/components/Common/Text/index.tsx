import React, { PureComponent } from 'react';
import {
  Text as EText,
  Icon as EIcon,
  IconProps as EIconProps,
  TextProps as ETextProps,
  withTheme,
  ThemeProps,
} from 'react-native-elements';
import { StyleSheet } from 'react-native';
import Font from '@themes/Font';
import { lightTheme } from '@themes/Theme';
import { Translation } from 'react-i18next';
import QuickView from '../View/QuickView';

type TFontFamily = typeof Font.fontFamily;
type TFontSize = typeof Font.fontSize;
type TType = typeof lightTheme.Text;

export interface TextProps extends Omit<ETextProps, 'fontFamily' | 'fontSize'> {
  fontFamily?: keyof TFontFamily;
  fontSize?: keyof TFontSize | number;
  type?: keyof TType;
  primary?: boolean;
  secondary?: boolean;
  success?: boolean;
  warning?: boolean;
  error?: boolean;
  color?: string;
  center?: boolean;
  iconRight?: boolean;
  margin?: number;
  marginTop?: number;
  marginBottom?: number;
  marginLeft?: number;
  marginRight?: number;
  marginHorizontal?: number;
  marginVertical?: number;
  thin?: boolean;
  bold?: boolean;
  italic?: boolean;
  underline?: boolean;
  children?: any;
  icon?: EIconProps;
  iconContainerStyle?: any;
  t?: string | Array<any>;
  theme?: any;
}

class Text extends PureComponent<TextProps> {
  static defaultProps = {
    iconRight: false,
    type: 'paragraph',
  };

  render() {
    const {
      fontFamily,
      fontSize,
      primary,
      secondary,
      success,
      error,
      warning,
      color,
      center,
      margin,
      marginLeft,
      marginRight,
      marginBottom,
      marginTop,
      marginHorizontal,
      marginVertical,
      underline,
      thin,
      bold,
      italic,
      icon,
      iconRight,
      iconContainerStyle,
      style,
      children,
      theme,
      type,
      t,
      ...otherProps
    } = this.props;

    /**
     * Color Handle
     */
    const { colors } = theme;
    // eslint-disable-next-line max-len
    let textColor = type ? theme.Text[type].color || theme.colors.primaryText : theme.colors.primaryText;
    if (primary) textColor = colors.primary;
    if (secondary) textColor = colors.secondary;
    if (success) textColor = colors.success;
    if (warning) textColor = colors.warning;
    if (error) textColor = colors.error;
    if (color) textColor = color;

    /**
     * textStyle
     */
    const textStyle = StyleSheet.flatten([
      type && theme.Text[type],
      fontFamily && {
        fontFamily: Font.fontFamily[fontFamily],
      },
      italic && {
        fontStyle: 'italic',
      },
      thin && {
        fontWeight: '200',
      },
      bold && {
        fontWeight: 'bold',
      },
      (fontSize && typeof fontSize === 'number') && {
        fontSize,
      },
      (fontSize && typeof fontSize === 'string') && {
        fontSize: Font.fontSize[fontSize],
      },
      center && {
        textAlign: 'center',
      },
      margin && { margin },
      marginTop && { marginTop },
      marginBottom && { marginBottom },
      marginLeft && { marginLeft },
      marginRight && { marginRight },
      marginHorizontal && { marginHorizontal },
      marginVertical && { marginVertical },
      underline && { textDecorationLine: 'underline' },
      { color: textColor },
      style,
    ]);

    /**
     * iconStyle
     */
    const iconStyle = StyleSheet.flatten([
      textStyle,
      { marginRight: iconRight ? 0 : 2 },
      { marginLeft: iconRight ? 2 : 0 },
      iconContainerStyle,
    ]);
    if (icon) {
      return (
        <QuickView testID="EIconText" row={!iconRight} rowReverse={iconRight} justifyContent={iconRight ? 'flex-end' : 'flex-start'}>
          <QuickView style={iconStyle}>
            <EIcon {...icon} name={icon.name} type={icon.type || 'material-community'} size={icon.size || 15} color={icon.color || textColor} />
          </QuickView>
          <QuickView>
            {
              (!children && t) ? (
                <Translation>
                  {
                    (trans) => <EText {...otherProps} style={textStyle}>{typeof t === 'string' ? trans(t) : trans(t[0], t[1])}</EText>
                  }
                </Translation>
              ) : (
                <EText {...otherProps} style={textStyle}>
                  {children}
                </EText>
              )
            }
          </QuickView>
        </QuickView>
      );
    }

    if (!children && t) {
      return (
        <Translation>
          {
            (trans) => <EText {...otherProps} style={textStyle}>{typeof t === 'string' ? trans(t) : trans(t[0], t[1])}</EText>
          }
        </Translation>
      );
    }
    return (
      <EText {...otherProps} style={textStyle}>
        {children}
      </EText>
    );
  }
}

export default withTheme(Text as React.ComponentType<TextProps & ThemeProps<any>>);
