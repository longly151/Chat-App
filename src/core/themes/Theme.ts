import { ThemeEnum } from '@contents/Config/redux/slice';
import { darkListItem, lightListItem } from './ThemeComponent/ListItem';
import { darkCalendar, lightCalendar } from './ThemeComponent/Calendar';
import { darkMap, lightMap } from './ThemeComponent/Map';
import { darkText, lightText } from './ThemeComponent/Text';
import { darkButton, lightButton } from './ThemeComponent/Button';
import { darkHeader, lightHeader } from './ThemeComponent/Header';
import { darkInput, lightInput } from './ThemeComponent/Input';
import { darkCheckBox, lightCheckBox } from './ThemeComponent/ListCheckBox';
import { darkModal, lightModal } from './ThemeComponent/Modal';
import { darkParallaxScrollView, lightParallaxScrollView } from './ThemeComponent/ParallaxScrollView';
import Color from './Color';

/**
 * Theme
 */
export const darkTheme = {
  key: ThemeEnum.DARK,
  colors: {
    primary: Color.darkPrimary,
    secondary: Color.darkSecondary,
    success: Color.darkSuccess,
    warning: Color.darkWarning,
    error: Color.darkError,

    /**
     * Component Color
     */
    primaryBackground: Color.darkPrimaryBackground,
    secondaryBackground: Color.darkSecondaryBackground,
    primaryText: Color.darkPrimaryText,
    secondaryText: Color.darkSecondaryText,
    loading: Color.darkPrimary,
  },

  /**
   * Component
   */
  Text: darkText,
  Button: darkButton,
  Header: darkHeader,
  Input: darkInput,
  CheckBox: darkCheckBox,
  Modal: darkModal,
  ParallaxScrollView: darkParallaxScrollView,
  Map: darkMap,
  Calendar: darkCalendar,
  ListItem: darkListItem,
};

export const lightTheme = {
  key: ThemeEnum.LIGHT,
  colors: {
    primary: Color.lightPrimary,
    secondary: Color.lightSecondary,
    success: Color.lightSuccess,
    warning: Color.lightWarning,
    error: Color.lightError,

    /**
     * Component Color
     */
    primaryBackground: Color.lightPrimaryBackground,
    secondaryBackground: Color.lightSecondaryBackground,
    primaryText: Color.lightPrimaryText,
    secondaryText: Color.lightSecondaryText,
    loading: Color.lightPrimary,
  },

  /**
   * Component
   */
  Text: lightText,
  Button: lightButton,
  Header: lightHeader,
  Input: lightInput,
  CheckBox: lightCheckBox,
  Modal: lightModal,
  ParallaxScrollView: lightParallaxScrollView,
  Map: lightMap,
  Calendar: lightCalendar,
  ListItem: lightListItem,
};
