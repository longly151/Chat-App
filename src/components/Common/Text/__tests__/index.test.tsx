/* eslint-disable no-undef */
import React from 'react';
import Font from '@themes/Font';
import TestHelper, { EnumComponent } from '@utils/testHelper';
import { lightTheme } from '@themes/Theme';
import { Platform } from 'react-native';
import Color from '@themes/Color';
import Text from '..';

const target = <Text />;
const type = EnumComponent.HOC;

describe('Text Component', () => {
  TestHelper.basicTest(target, type);

  it('should apply all style-related props', () => {
    const props = {
      margin: 10,
      marginVertical: 10,
      marginHorizontal: 10,
      marginTop: 10,
      marginBottom: 10,
      marginLeft: 10,
      marginRight: 10,
    };
    TestHelper.testStyleProps(target, props, type);
  });

  it('should apply all custom props', () => {
    /**
     * center
     */
    const testStyle = TestHelper.getStyle(target, { center: true }, type);
    expect(testStyle.textAlign).toBe('center');

    /**
     * fontFamily, fontWeight, fontSize
     */
    const props2 = {
      fontFamily: 'GoogleSans',
      thin: true,
      fontSize: 'large',
    };
    const testStyle2 = TestHelper.getStyle(target, props2, type);
    expect(testStyle2.fontFamily).toBe(Font.fontFamily.GoogleSans);
    expect(testStyle2.fontWeight).toBe('200');
    expect(testStyle2.fontSize).toBe(Font.fontSize.large);

    /**
     * primary, secondary, success,  warning, error, color,
     */
    Platform.OS = 'android';
    const testStyle3 = TestHelper.getStyle(target, { primary: true }, type);
    expect(testStyle3.color).toBe(lightTheme.colors.primary);
    const testStyle4 = TestHelper.getStyle(target, { secondary: true }, type);
    expect(testStyle4.color).toBe(lightTheme.colors.secondary);
    const testStyle5 = TestHelper.getStyle(target, { success: true }, type);
    expect(testStyle5.color).toBe(lightTheme.colors.success);
    const testStyle6 = TestHelper.getStyle(target, { error: true }, type);
    expect(testStyle6.color).toBe(lightTheme.colors.error);
    const testStyle7 = TestHelper.getStyle(target, { warning: true }, type);
    expect(testStyle7.color).toBe(lightTheme.colors.warning);
    const testStyle8 = TestHelper.getStyle(target, { color: Color.orange }, type);
    expect(testStyle8.color).toBe(Color.orange);
    jest.resetModules();

    /**
     * underline, bold, italic
     */
    const testStyle15 = TestHelper.getStyle(target, { underline: true }, type);
    expect(testStyle15.textDecorationLine).toBe('underline');
    const testStyle16 = TestHelper.getStyle(target, { bold: true }, type);
    expect(testStyle16.fontWeight).toBe('bold');
    const testStyle17 = TestHelper.getStyle(target, { italic: true }, type);
    expect(testStyle17.fontStyle).toBe('italic');

    /**
     * icon
     */
    // const component = mount(<Text icon={{ name: 'account' }} />);
    // const demo = component.findWhere((node) => node.prop('testID') === 'EIconText');
    // expect(demo.length).toBeGreaterThanOrEqual(1);

    const props3 = TestHelper.getComponent(target, { icon: { name: 'account' } }, type).props();
    expect(props3.testID).toBe('EIconText');
    expect(props3.row).toBe(true);
    expect(props3.rowReverse).not.toBe(true);

    /**
     * iconRight
     */
    const props4 = TestHelper.getComponent(target, { icon: { name: 'account' }, iconRight: true }, type).props();
    expect(props3.testID).toBe('EIconText');
    expect(props4.row).not.toBe(true);
    expect(props4.rowReverse).toBe(true);

    /**
     * iconContainerStyle
     */
    const props5 = TestHelper.getComponent(target, { icon: { name: 'account' }, iconContainerStyle: { color: Color.orange } }, type).childAt(0).props();
    expect(props3.testID).toBe('EIconText');
    expect(props5.style).toMatchObject({ color: Color.orange });

    /**
     * type
     */
    const testStyle18 = TestHelper.getStyle(target, { type: 'xTitle' }, type);
    expect(testStyle18).toMatchObject(lightTheme.Text.xTitle);
  });
});
