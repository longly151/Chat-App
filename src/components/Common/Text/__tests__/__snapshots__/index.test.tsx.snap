// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component should render without issues 1`] = `
<Themed.Text
  style={
    Object {
      "color": "#121212",
      "fontSize": 14,
    }
  }
/>
`;
